redis/cluster.py

```diff
@@ -0,0 +1,1249 @@
+        ], parse_pubsub_channels),
+        list_keys_to_dict([
+            "PUBSUB NUMSUB",
+        ], parse_pubsub_numsub),
+        list_keys_to_dict([
+            "PUBSUB NUMPAT",
+        ], parse_pubsub_numpat),
+        list_keys_to_dict([
+            "KEYS",
+        ], merge_result),
+    )
+
+    def __init__(
+            self,
+            host=None,
+            port=6379,
+            startup_nodes=None,
+            cluster_error_retry_attempts=3,
+            skip_full_coverage_check=False,
+            reinitialize_steps=10,
+            read_from_replicas=False,
+            url=None,
+            **kwargs
+    ):
+        """
+        :startup_nodes:
+            List of nodes from which initial bootstrapping can be done
+        :host:
+            Can be used to point to a startup node
+        :port:
+            Can be used to point to a startup node
+        :skip_full_coverage_check:
+            Skips the check of cluster-require-full-coverage config, useful for
+            clusters without the CONFIG command (like aws)
+       :read_from_replicas:
+            Enable read from replicas in READONLY mode. You can read possibly
+            stale data.
+            When set to true, read commands will be assigned between the
+            primary and its replications in a Round-Robin manner.
+        :cluster_error_retry_attempts:
+
+        :**kwargs:
+            Extra arguments that will be sent into Redis instance when created
+            (See Official redis-py doc for supported kwargs
+        [https://github.com/andymccurdy/redis-py/blob/master/redis/client.py])
+            Some kwargs are not supported and will raise a
+            RedisClusterException:
+                - db (Redis do not support database SELECT in cluster mode)
+        """
+        if "db" in kwargs:
+            # Argument 'db' is not possible to use in cluster mode
+            raise RedisClusterException(
+                "Argument 'db' is not possible to use in cluster mode"
+            )
+
+        # Get the startup node/s
+        from_url = False
+        if not startup_nodes:
+            startup_nodes = []
+        if host and port:
+            startup_nodes.append(ClusterNode(host, port))
+        elif url:
+            from_url = True
+            url_options = parse_url(url)
+            if "path" in url_options:
+                raise RedisClusterException(
+                    "RedisCluster does not currently support Unix Domain "
+                    "Socket connections")
+            if "db" in url_options and url_options["db"] != 0:
+                # Argument 'db' is not possible to use in cluster mode
+                raise RedisClusterException(
+                    "A ``db`` querystring option can only be 0 in cluster mode"
+                )
+            kwargs.update(url_options)
+            startup_nodes.append(ClusterNode(kwargs['host'], kwargs['port']))
+        if not startup_nodes:
```

Comment:
I prefer the explicit syntax:
```suggestion
        if startup_nodes == []:
```

Alternatively a check on len(startup_nodes) == 0 would be nice. It helps juniors understand the code base, while achieving the same value.

Reply:
changed to len(startup_nodes) == 0